### PR TITLE
Added updatecli and Canal update automation.

### PR DESCRIPTION
**Pull request by rbrtbnfgl**
This PR adds updatecli for version update automation with the update for the Canal charts.
The update for the Canal chart checks if a new image is released for flannel (https://github.com/rancher/image-build-flannel) and Calico (https://github.com/rancher/image-build-calico) and updates the image tag version.
`packageVersion` and the chart `version` are updated accordingly.

### CODE REVIEW

**Review by thomasferrandiz — APPROVED** [review]

File: updatecli/updatecli.d/updatecanal.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+name: "Update Canal version" 
+
+sources:
+  flannel:
+    name: Get flannel version
+    kind: githubrelease
+    spec:
+      owner: rancher
+      repository: image-build-flannel
+      token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
```
Comment (manuelbuil) [inline updatecli/updatecli.d/updatecanal.yaml:11]: Try using this instead https://github.com/rancher/image-build-cni-plugins/blob/main/updatecli/updatecli.d/updateplugins.yaml#L11

File: updatecli/updatecli.d/updatecanal.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      username: '{{ requiredEnv "UPDATECLI_GITHUB_ACTOR" }}'
```
Comment (manuelbuil) [inline updatecli/updatecli.d/updatecanal.yaml:12]: I don't think we need username here

File: updatecli/updatecli.d/updatecanal.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        - name: CALICO_VERSION
+          value: '{{ source "calico" }}'
+
+
+scms:
+  default:
+    kind: github
+    spec:
+      token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
+      username: '{{ requiredEnv "UPDATECLI_GITHUB_ACTOR" }}'
```
Comment (manuelbuil) [inline updatecli/updatecli.d/updatecanal.yaml:70]: Use this instead: https://github.com/rancher/image-build-cni-plugins/blob/main/updatecli/updatecli.d/updateplugins.yaml#L45-L46, that way we use values.yaml

File: updatecli/updatecli.d/updatecli.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
+# This small test makes sure that updatecli is working properly on a repo.
```
Comment (manuelbuil) [inline updatecli/updatecli.d/updatecli.yml:1]: Do you think we need this yaml file, I thought it was just for testing

**Review by manuelbuil — CHANGES_REQUESTED** [review]

**Review by manuelbuil — APPROVED** [review]

**Pull request merged**